corehq/apps/campaign/static/campaign/js/dashboard.js

```diff
@@ -246,10 +246,10 @@ var onHideWidgetModal = function () {
 
 var onShowWidgetModal = function (event) {
     const triggerSource = event.relatedTarget;
-    if (triggerSource.id === 'edit-widget-btn') {
-        $modalTitleElement.text(editWidgetText);
-    } else {
+    if ($(triggerSource).data('source') === 'add-widget-dropdown') {
```

Comment:
Can we access the action to decide this, which is already available, instead of adding a new data attr?